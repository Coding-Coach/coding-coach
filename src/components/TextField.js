import React from 'react';
import PropTypes from 'prop-types';

class TextField extends React.Component {
  render() {
    const {
      value,
      onChange,
      onFocus,
      onBlur,
      label,
      placeholder,
      type,
      multiline,
      classes = defaultClasses
    } = this.props;

    const { root, input, labelStyle } = classes;
    const _renderTextInput = () => {
      if (!multiline) {
        return (
          <div className={root}>
            <label className={labelStyle}>{label}</label>
            <input
              className={input}
              label={label}
              onChange={onChange}
              onBlur={onBlur}
              onFocus={onFocus}
              type={type}
              value={value}
              placeholder={placeholder}
            />
          </div>
        );
      } else {
        return (
          <div className={root}>
            <label className={labelStyle}>{label}</label>
            <textarea
              className={input}
              label={label}
              onChange={onChange}
              onBlur={onBlur}
              onFocus={onFocus}
              type={type}
              value={value}
              placeholder={placeholder}
            />
          </div>
        );
      }
    };
    return _renderTextInput();
  }
}

const defaultClasses = {
  root: 'textfield__root',
  input: 'textfield__input',
  labelStyle: 'textfield__labelStyle'
};

TextField.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  multiline: PropTypes.bool
};

export default TextField;
