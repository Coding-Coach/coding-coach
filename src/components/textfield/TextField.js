import React from 'react';
import classnames from 'classnames';
import { func, string } from 'prop-types';

export default function TextField({ label, name, onChange, type, value, error }) {
  const inputClassNames = classnames(
    `
      block
      border rounded-sm
      leading-tight
      border-secondary-lighter bg-white
      w-full p-2
      focus:outline-none focus:border-primary focus:text-primary
    `,
    {
      'border-danger': !!error,
    },
  );

  return (
    <div className="mb-4 text-secondary-light focus-within:text-primary">
      {label && (
        <label className={classnames('block mb-1', { 'text-danger': !!error })} htmlFor={name}>
          {label}
        </label>
      )}

      <input
        type={type}
        className={inputClassNames}
        name={name}
        value={value}
        onChange={onChange}
      />

      {!!error && <div className="mt-1 text-danger text-sm">{error}</div>}
    </div>
  );
}

TextField.propTypes = {
  label: string,
  name: string,
  onChange: func,
  type: string,
  value: string,
  error: string,
};

TextField.defaultProps = {
  type: 'text',
};
