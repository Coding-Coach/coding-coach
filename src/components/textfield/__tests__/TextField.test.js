import React from 'react';
import { render } from 'react-testing-library';
import TextField from '../TextField';

describe('components/textfield/TextField', () => {
  it('should render the textfield', () => {
    const { container } = render(<TextField />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the label', () => {
    const { getByText } = render(<TextField label="Testing Label" />);

    // Check if is present
    expect(getByText('Testing Label')).toBeTruthy();
  });
});
