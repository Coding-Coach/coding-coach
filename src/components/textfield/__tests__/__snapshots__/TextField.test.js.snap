// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/textfield/TextField should render the textfield 1`] = `
<div
  class="mb-4"
>
  <input
    class="block border border-secondary-lighter bg-white w-full p-2 rounded-sm focus:outline-none focus:border focus:border-primary"
    type="text"
    value=""
  />
</div>
`;
